core/src/main/kotlin/cash/grammar/kotlindsl/model/gradle/DependencyContainer.kt

```diff
@@ -18,26 +19,15 @@ public class DependencyContainer(
   public val elements: List<Any>,
```

Comment:
To avoid `Any` all together, what do you think if we introduce `DependencyElement` so this will be come list<DependencyElement>?

```
public sealed class DependencyElement(protected open val statement: StatementContext)

public data class DependencyDeclarationElement(
  val declaration: DependencyDeclaration,
  override val statement: StatementContext
) : DependencyElement(statement)

public data class OtherElement(override val statement: StatementContext) : DependencyElement(statement)
```

I can refactor this as a follow up PR.

Reply:
I think that sounds ok. I think all the properties of the `DependencyElement` should probably be `public`, right?

Reply:
Done in 65b74c1e1c5faf2fd460be5466fa1499a08e01d6